bitnami/kafka/templates/scripts-configmap.yaml

```
@@ -109,12 +109,28 @@ data:
     export KAFKA_CFG_ADVERTISED_LISTENERS="INTERNAL://${MY_POD_NAME}.{{ $fullname }}-headless.{{ $releaseNamespace }}.svc.{{ $clusterDomain }}:{{ $interBrokerPort }},CLIENT://${MY_POD_NAME}.{{ $fullname }}-headless.{{ $releaseNamespace }}.svc.{{ $clusterDomain }}:{{ $clientPort }},EXTERNAL://${EXTERNAL_ACCESS_IP}:${EXTERNAL_ACCESS_PORT}"
     {{- end }}
     {{- end }}
-
+    
     {{- if (include "kafka.tlsEncryption" .) }}
-    if [[ -f "/certs/kafka.truststore.jks" ]] && [[ -f "/certs/kafka-${ID}.keystore.jks" ]]; then
+    JKS_TRUSTSTORE_PATH="/certs"
+    {{- if .Values.auth.jksTruststoreSecret }}
+    JKS_TRUSTSTORE_PATH="/truststore"
+    {{- end }}
+
+    JKS_TRUSTSTORE="${JKS_TRUSTSTORE_PATH}/kafka.trustore.jks"
+    JKS_KEYSTORE="/certs/kafka-${ID}.keystore.jks"
+    
+    {{- if .Values.auth.jksTruststore }}
+    JKS_TRUSTSTORE="${JKS_TRUSTSTORE_PATH}/{{ .Values.auth.jksTruststore }}"
+    {{- end }}    
+    
+    {{- if .Values.auth.jksKeystoreSAN }}
+    JKS_KEYSTORE="/certs/{{ .Values.auth.jksKeystoreSAN }}"
+    {{- end }}
```

```suggestion
    {{- if .Values.auth.jksTruststoreSecret }}
    JKS_TRUSTSTORE="/truststore/kafka.trustore.jks"
    {{- else if and .Values.auth.jksTruststoreSecret .Values.auth.jksTruststore }}
    JKS_TRUSTSTORE="truststore/{{ .Values.auth.jksTruststore }}"
    {{- else }}
    JKS_TRUSTSTORE="/cert/kafka.trustore.jks"
    {{- end }}

    {{- if .Values.auth.jksKeystoreSAN }}
    JKS_KEYSTORE="/certs/{{ .Values.auth.jksKeystoreSAN }}"
    {{- else }}
    JKS_KEYSTORE="/certs/kafka-${ID}.keystore.jks"
    {{- end }}
```

How do you see doing the logic more explicit? So we can see what is happening depending on the parameters and also we avoid one extra variable (the one with the path)

---

I like your suggestion.  Please review the small fix I did in the next comment.